src/main/java/book/store/service/impl/CategoryServiceImpl.java

```diff
@@ -24,18 +24,23 @@ public CategoryDto save(CategoryDto categoryDto) {
 
     @Override
     public List<CategoryDto> findAll(Pageable pageable) {
-        return categoryRepository.findAll(pageable).stream().map(categoryMapper::toDto).toList();
+        return categoryRepository.findAll(pageable)
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
     }
 
     @Override
     public CategoryDto getById(Long id) {
-        return categoryMapper.toDto(categoryRepository.findById(id).get());
+        return categoryMapper.toDto(categoryRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Can't find category with ID " + id)));
     }
 
     @Override
     public CategoryDto update(Long id, CategoryDto categoryDto) {
-        Category category = categoryMapper.toModel(categoryDto);
-        category.setId(id);
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Can't find category with ID " + id));
```

Comment:
please use EntityNotFoundException for such cases. Avoid overusing generic exceptions like runtime exception